internal/component/prometheus/operator/common/component_test.go

```diff
@@ -8,48 +8,42 @@ import (
 	"time"
 
 	"github.com/go-kit/log"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/storage"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/atomic"
+
 	"github.com/grafana/alloy/internal/component"
 	"github.com/grafana/alloy/internal/component/prometheus/operator"
 	"github.com/grafana/alloy/internal/service/cluster"
 	http_service "github.com/grafana/alloy/internal/service/http"
 	"github.com/grafana/alloy/internal/service/labelstore"
 	"github.com/grafana/alloy/internal/util"
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/prometheus/config"
-	"github.com/prometheus/prometheus/storage"
-	"github.com/stretchr/testify/require"
-	"go.uber.org/atomic"
 )
 
 type crdManagerFactoryHungRun struct {
-	stopRun chan struct{}
-	onRun   chan struct{}
+	running         *atomic.Bool
+	contextCenceled *atomic.Bool
+	stopRun         chan struct{}
 }
 
-func (m crdManagerFactoryHungRun) New(_ component.Options, _ cluster.Cluster, _ log.Logger,
-	_ *operator.Arguments, _ string, _ labelstore.LabelStore) crdManagerInterface {
-
-	return &crdManagerHungRun{
-		onRun:   m.onRun,
-		stopRun: m.stopRun,
-	}
+func (m crdManagerFactoryHungRun) New(_ component.Options, _ cluster.Cluster, _ log.Logger, _ *operator.Arguments, _ string, _ labelstore.LabelStore) crdManagerInterface {
+	return &crdManagerHungRun{m.running, m.contextCenceled, m.stopRun}
 }
 
 type crdManagerHungRun struct {
-	stopRun chan struct{}
-	onRun   chan struct{}
+	running         *atomic.Bool
+	contextCenceled *atomic.Bool
```

Comment:
```suggestion
	contextCancelled *atomic.Bool
```

Reply:
So you can spell it either _Cancelled_ or _Canceled_. In golang context package Canceled is used so I went with that.